build: Bump minimum Qt version to 5.11.3

hebasto opened the pull request:
The current minimum Qt version is 5.9.5 which has been set in bitcoin/bitcoin#21286.

Distro support:
- centos 7 -- unsupported since bitcoin/bitcoin#23511
- centos 8 -- [5.15.2](http://mirror.centos.org/centos/8/AppStream/x86_64/os/Packages/qt5-qtbase-5.15.2-3.el8.x86_64.rpm)
- buster -- [5.11.3](https://packages.debian.org/buster/libqt5core5a)
- bullseye  -- [5.15.2](https://packages.debian.org/bullseye/libqt5core5a)
- _bionic_ -- [5.9.5](https://packages.ubuntu.com/bionic/libqt5core5a)
- focal -- [5.12.8](https://packages.ubuntu.com/focal/libqt5core5a)

As another Ubuntu LTS is coming soon, it seems unreasonable to stick to Qt 5.9 which support [ended](https://www.qt.io/blog/2017/06/07/renewed-qt-support-services) on 2020-05-31. Anyway, it's still possible to build Bitcoin Core GUI with depends on bionic system.

Bumping the minimum Qt version allows to make code safer and more reliable, e.g.:
- functor-parameter overload of [`QMetaObject::invokeMethod`](https://doc.qt.io/qt-5/qmetaobject.html#invokeMethod-4)
- fixed https://bugreports.qt.io/browse/QTBUG-10907

An example of the patch using the functor-overload of `QMetaObject::invokeMethod`:
```diff
--- a/src/qt/walletmodel.cpp
+++ b/src/qt/walletmodel.cpp
@@ -349,7 +349,7 @@ bool WalletModel::changePassphrase(const SecureString &oldPass, const SecureStri
 static void NotifyUnload(WalletModel* walletModel)
 {
     qDebug() << "NotifyUnload";
-    bool invoked = QMetaObject::invokeMethod(walletModel, "unload");
+    bool invoked = QMetaObject::invokeMethod(walletModel, &WalletModel::unload);
     assert(invoked);
 }
 
```
It uses the same new syntax as signal-slot connection with compile-time check. Also see bitcoin/bitcoin#16348.


This PR is intended to be merged early [after](https://github.com/bitcoin/bitcoin/issues/22969) branching `23.x` off.

shaavan commented:
I am concept ACK on upgrading the minimum required Qt version to a later version because, as it is mentioned, it makes the code more secure and reliable.

However, I have one question I would like to ask. As you mentioned:
> it seems unreasonable to stick to Qt 5.9, which support ended on 2020-05-31

I agree with the reasoning. However, when I checked for the Standard Support for **Qt 5.11.3**, I noticed it ended on **2019-05-22**.
So if we are upgrading the minimum version, why are we choosing the version whose support ended almost two years ago?

fanquake reviewed (commented):
> functor-parameter overload of QMetaObject::invokeMethod
> fixed https://bugreports.qt.io/browse/QTBUG-10907

~0. If this is all the motivation, that seems fairly weak.

MarcoFalke commented:
> ~0. If this is all the motivation, that seems fairly weak.

Not sure what the `invokeMethod` will change in practice, but when it changes runtime errors into compile time errors (?), then it seems like a fine motivation. With 0.24 it should be fine to drop bionic-qt.

hebasto commented:
@fanquake 
> > functor-parameter overload of QMetaObject::invokeMethod
> > fixed https://bugreports.qt.io/browse/QTBUG-10907
> 
> ~0. If this is all the motivation, that seems fairly weak.

@MarcoFalke 
> > ~0. If this is all the motivation, that seems fairly weak.
> 
> Not sure what the `invokeMethod` will change in practice, but when it changes runtime errors into compile time errors (?), then it seems like a fine motivation. With 0.24 it should be fine to drop bionic-qt.

PR description has been updated with an example. Also see bitcoin/bitcoin#16348 by @promag.

hebasto committed:
ci: Switch from bionic to buster

This change is a prerequisite for the following bumping Qt minimum
version to 5.11.3. It is required as bionic has Qt 5.9.5.

Effectively, this also changes:
- gcc from 8.4.0 to 8.3.0
- python from 3.6.5 to 3.7.3

hebasto committed:
build: Bump minimum Qt version to 5.11.3

hebasto commented:
Updated e39a9da025e07f1ca33e28c934b29b3393a451a0 -> 956f7322f60db7b8be551c9074b4c633e514079d ([pr24132.01](https://github.com/hebasto/bitcoin/commits/pr24132.01) -> [pr24132.02](https://github.com/hebasto/bitcoin/commits/pr24132.02)):
- rebased
- addressed @fanquake's [comment](https://github.com/bitcoin/bitcoin/pull/24132#discussion_r791337606)

hebasto commented on ci/test/00_setup_env_native_qt5.sh:
```diff
@@ -7,7 +7,7 @@
 
 export CONTAINER_NAME=ci_native_qt5
-export DOCKER_NAME_TAG=ubuntu:18.04  # Check that bionic gcc-8 can compile our C++17 and run our functional tests in python3, see doc/dependencies.md
+export DOCKER_NAME_TAG=debian:buster  # Check that buster gcc-8 can compile our C++17 and run our functional tests in python3, see doc/dependencies.md
```
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/24132#issuecomment-1030706408).

MarcoFalke commented:
cr ACK 956f7322f60db7b8be551c9074b4c633e514079d

The python3 CI change should be fine, as we still test python3.6 in the bionic nowallet task.

laanwj commented:
Concept ACK

promag reviewed (commented):
Concept ACK.

Then we can also ditch `GUIUtil::ObjectInvoke`.

gruve-p commented:
Concept ACK

fanquake reviewed (approved):
ACK 956f7322f60db7b8be551c9074b4c633e514079d
